# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/layers/client/CanvasClient.cpp
# Commit: cdaa8a4e9e36
# Full Hash: cdaa8a4e9e366ad041b429a8f5bd0b9b259d4722
# Author: Jeff Gilbert <jgilbert@mozilla.com>
# Date: 2020-06-09 09:21:34
# Regressor Bug: 1632249
# File Overlap Count: 1
# Description:
#   Bug 1632249 - NON_PREMULT TextureClient iff NON_PREMULT CanvasClient. r=handyman
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D78800
# ==============================================================================

diff -r 9ff26bcc580c -r cdaa8a4e9e36 gfx/layers/client/CanvasClient.cpp
--- a/gfx/layers/client/CanvasClient.cpp	Mon Jun 08 20:35:38 2020 +0000
+++ b/gfx/layers/client/CanvasClient.cpp	Mon Jun 08 20:58:03 2020 +0000
@@ -31,6 +31,12 @@
 void CanvasClient::UseTexture(TextureClient* const aTexture) {
   MOZ_ASSERT(aTexture);
 
+  const auto isClientNonPremult =
+      bool(mTextureFlags & TextureFlags::NON_PREMULTIPLIED);
+  const auto isTextureNonPremult =
+      bool(aTexture->GetFlags() & TextureFlags::NON_PREMULTIPLIED);
+  MOZ_ALWAYS_TRUE(isTextureNonPremult == isClientNonPremult);
+
   bool changed = false;
 
   if (aTexture != mFrontBuffer) {
